Core/URLExtension.swift

```diff
@@ -20,8 +20,15 @@
 
 import Foundation
 
+
+
 extension URL {
     
+    enum URLProtocol: String {
+        case http
+        case https
+    }
+    
     private static let webUrlRegex = "^(https?:\\/\\/)?([\\da-z\\.-]+\\.[a-z\\.]{2,6}|(([\\d]+[.]){3}[\\d]+))([\\/]?[\\/:?=&#]{1}[\\%\\da-zA-Z_\\.-]+)*[\\/\\?]?$"
```

Comment:
Could use https constant from the enum here (like is done in the appendScheme func)